app/models/compliance/ObligationDetail.scala

```diff
@@ -18,13 +18,15 @@ package models.compliance
 
 import play.api.libs.json.{Json, OFormat}
 
+import java.time.LocalDate
+
 case class ObligationDetail(
-                              status: String,
-                              inboundCorrespondenceFromDate: String,
-                              inboundCorrespondenceToDate: String,
-                              inboundCorrespondenceDateReceived: Option[String],
-                              inboundCorrespondenceDueDate: String,
-                              periodKey: String
+                             status: ComplianceStatusEnum.Value,
+                             inboundCorrespondenceFromDate: LocalDate,
+                             inboundCorrespondenceToDate: LocalDate,
+                             inboundCorrespondenceDateReceived: Option[LocalDate],
+                             inboundCorrespondenceDueDate: LocalDate,
+                             periodKey: String
```

Comment:
As per the schema , the property has a min and max length, which I guess needs to be checked.
                "periodKey":{
                           "type":"string",
                           "minLength":1,
                           "maxLength":4
                        }

Reply:
We don't really worry about the period key, I don't see it ever being used (it is purely for JSON parsing reasons)